src/index.js

```diff
@@ -14,7 +15,44 @@ module.exports = Class.extend({
          throw new Error('This plugin must be used with AWS');
       }
 
-      throw new Error('This plugin\'s functionality has not been implemented');
+      this.hooks = {
+         'aws:package:finalize:mergeCustomProviderResources': this._modifyTemplate.bind(this),
+      };
+   },
+
+   _modifyTemplate: function() {
+      var self = this,
+          stackTags = this._getStackTags(),
+          template = this._serverless.service.provider.compiledCloudFormationTemplate;
+
+      _.chain(template.Resources)
+         .filter({ Type: 'AWS::CloudFront::Distribution' })
+         .each(function(resource) {
+            this._addTagsToResource(resource, stackTags);
+         }.bind(this));
+   },
+
+   _getStackTags: function() {
+      return _.extend({
+         STAGE: this._provider.getStage(), // The provider's stackTags does not include the stage
+      }, this._serverless.service.provider.stackTags);
```

Comment:
Not quite sure how I missed this in testing earlier, was `this._service.provider.stackTags`. Think I had an uncommitted change.